tests/test_steps_edge.py

```
@@ -8,104 +15,271 @@
+        mock_st.info.assert_called()
+
+
+def test_add_score_column_clash_logic():
+    """Verify that Add Score Column handles existing names correctly."""
+    df = pd.DataFrame({"Name": ["A"], "Score1": [1.0], "Score2": [2.0]})
+    mock_state = MagicMock()
+    mock_state.manual_df = df
+    with patch("src.ui.steps.st") as mock_st:
+        mock_st.session_state = mock_state
+        mock_st.file_uploader.return_value = None
+        mock_st.button.side_effect = lambda label, **_: label == "➕ Add Score Column"
```

Ruff (0.15.12)

[warning] String contains ambiguous `➕` (HEAVY PLUS SIGN). Did you mean `+` (PLUS SIGN)?

(RUF001)